homeassistant/components/teslemetry/climate.py

```diff
@@ -153,3 +170,121 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
+
+        # Scopes
+        self.scoped = Scope.VEHICLE_CMDS in scopes
+        if not self.scoped:
+            self._attr_supported_features = ClimateEntityFeature(0)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+
+        if (state := self.get("climate_state_cabin_overheat_protection")) is None:
+            self._attr_hvac_mode = None
+        else:
+            self._attr_hvac_mode = COP_MODES.get(state)
+
+        if (level := self.get("climate_state_cop_activation_temperature")) is None:
+            self._attr_target_temperature = None
+        else:
+            self._attr_target_temperature = COP_LEVELS.get(level)
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+        await self.async_set_hvac_mode(HVACMode.COOL)
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+        self.raise_for_scope()
+
+        if temp := kwargs.get(ATTR_TEMPERATURE):
+            if temp == 30:
+                cop_mode = CabinOverheatProtectionTemp.LOW
+            elif temp == 35:
+                cop_mode = CabinOverheatProtectionTemp.MEDIUM
+            elif temp == 40:
+                cop_mode = CabinOverheatProtectionTemp.HIGH
+            else:
+                raise ServiceValidationError(
+                    "Cabin overheat protection does not support that temperature",
+                    translation_domain=DOMAIN,
+                    translation_key="invalid_cop_temp",
+                )
+
+            await self.wake_up_if_asleep()
+            await self.handle_command(self.api.set_cop_temp(cop_mode))
+            self._attr_target_temperature = temp
+
+        if mode := kwargs.get(ATTR_HVAC_MODE):
+            # Set HVAC mode will call write_ha_state
+            await self.async_set_hvac_mode(mode)
+        else:
+            self.async_write_ha_state()
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set the climate mode and state."""
+        self.raise_for_scope()
+        await self.wake_up_if_asleep()
+        if hvac_mode == HVACMode.OFF:
+            await self.handle_command(
+                self.api.set_cabin_overheat_protection(on=False, fan_only=False)
+            )
+        elif hvac_mode == HVACMode.COOL:
+            await self.handle_command(
+                self.api.set_cabin_overheat_protection(on=True, fan_only=False)
+            )
+        elif hvac_mode == HVACMode.FAN_ONLY:
+            await self.handle_command(
+                self.api.set_cabin_overheat_protection(on=True, fan_only=True)
+            )
+
+        self._attr_hvac_mode = hvac_mode
```

Comment:
Perhaps move this to a sub-function to avoid calling `raise_for_scope` and `wake_up_if_asleep` twice (if called from `set_temperature`.